src/olympia/reviewers/views.py

```diff
@@ -649,14 +651,43 @@ def review(request, addon, channel=None):
     channel = amo.CHANNEL_CHOICES_LOOKUP.get(
         channel, amo.RELEASE_CHANNEL_LISTED)
 
-    if content_review_only and not acl.action_allowed(
-            request, amo.permissions.ADDONS_CONTENT_REVIEW):
-        raise PermissionDenied
+    # We're going to need those later:
+    was_auto_approved = (
+        channel == amo.RELEASE_CHANNEL_LISTED and
+        addon.current_version and addon.current_version.was_auto_approved)
+    unlisted_only = (
+        channel == amo.RELEASE_CHANNEL_UNLISTED or
+        not addon.has_listed_versions())
+
+    # If we're just looking (GET) we can bypass the specific permissions checks
+    # if we have ReviewerTools:View.
+    bypass_more_specific_permissions_because_read_only = (
+        request.method == 'GET' and acl.action_allowed(
+            request, amo.permissions.REVIEWER_TOOLS_VIEW))
+
+    if not bypass_more_specific_permissions_because_read_only:
+        # Are we looking at an unlisted review page, or (weirdly) the listed
+        # review page of an unlisted-only add-on?
+        if unlisted_only and not acl.check_unlisted_addons_reviewer(request):
+            raise PermissionDenied
+
+        # If we're only doing a content review, we just need to check for the
+        # content review permission, otherwise it's the "main" review page.
+        if content_review_only:
+            if not acl.action_allowed(
+                    request, amo.permissions.ADDONS_CONTENT_REVIEW):
+                raise PermissionDenied
+        else:
+            # Was the add-on auto-approved?
```

Comment:
What's the logic behind allowing someone with ADDONS_REVIEW permission to access the review page until the first webextension is submitted and then denying them from them onwards?  (Not saying it's wrong, just makes me wonder)

Reply:
Hmm.  Maybe someone with ADDONS_REVIEW should get what REVIEWER_TOOLS_VIEW gets - a read-only view?

Reply:
Well, by the time the webextension has been submitted and auto-approved, it's suddenly supposed to be a post-review...